OpenTrons/opentrons-api
# feat(robot-server): implement data files auto-deletion

Closes AUTH-467

# Overview

Adds auto-deletion of old data files. If the data files stored on disk exceed 50 files, then we start auto-deleting files as needed. The oldest file which is not being referenced by any analysis or run in the database is deleted from disk and removed from the database

## Test Plan and Hands on Testing

- Add 50 data files to the robot, use some of them in a run
- Check that 50 files still exist
- Add another file
- Check that the oldest unused file is deleted
(pro-tip: you can change the file limit in robot settings for testing)

## Changelog

- added `DataFileAutoDeleter` and `DataFileDeletionPlanner`
- updated data file router to use auto-deletion before adding a new file
- added new `remove()` method to `DataFilesStore`

## Review requests

- Usual code review

## Risk assessment

Medium. Doesn't change existing infrastructure, except that if implemented incorrectly, could delete data files unexpectedly.
